pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -129,31 +132,44 @@ func (i *OsqueryInstance) Healthy() error {
 		return errors.New("instance not started")
 	}
 
-	for _, srv := range i.extensionManagerServers {
-		serverStatus, err := srv.Ping(context.TODO())
-		if err != nil {
-			return fmt.Errorf("could not ping extension server: %w", err)
+	resultsChan := make(chan error)
+	gowrapper.Go(context.TODO(), i.slogger, func() {
+		for _, srv := range i.extensionManagerServers {
+			serverStatus, err := srv.Ping(context.TODO())
+			if err != nil {
+				resultsChan <- fmt.Errorf("could not ping extension server: %w", err)
+				return
+			}
+			if serverStatus.Code != 0 {
+				resultsChan <- fmt.Errorf("ping extension server returned %d: %s", serverStatus.Code, serverStatus.Message)
+				return
+			}
 		}
-		if serverStatus.Code != 0 {
-			return fmt.Errorf("ping extension server returned %d: %s",
-				serverStatus.Code,
-				serverStatus.Message)
 
+		clientStatus, err := i.extensionManagerClient.Ping()
+		if err != nil {
+			resultsChan <- fmt.Errorf("could not ping osquery extension client: %w", err)
+			return
+		}
+		if clientStatus.Code != 0 {
+			resultsChan <- fmt.Errorf("ping extension client returned %d: %s", clientStatus.Code, clientStatus.Message)
+			return
```

Comment:
Do you think it'd be worth putting this in a separate `gowrapper.Go` call for increased speed/parallelization, or nah?

Reply:
so i got the impression (but totally could have misunderstood) that the Pings from `i.extensionManagerServers` should be extremely quick and hadn't really considered further optimization here since we'll have to wait for this anyway. i'm happy to break out if you think it would add value, it only seems slightly more complicated to me at first glance

Reply:
Seems fine to leave it as-is!